internal/pkg/mqtt/client_test.go

```diff
@@ -897,9 +894,11 @@ func mockUnmarshalerError([]byte, interface{}) error {
 }
 
 // receiveMessage polls the provided channel until the expected number of messages has been received.
-func receiveMessage(group *sync.WaitGroup, messageChannel <-chan types.MessageEnvelope, expectedMessages int) {
+func receiveMessage(t *testing.T, group *sync.WaitGroup, topicChannel types.TopicChannel, expectedMessages int) {
 	for counter := 0; counter < expectedMessages; counter++ {
-		<-messageChannel
+		message := <-topicChannel.Messages
+		// Unit test is using simple topic, i.e. without wild card
+		assert.Equal(t, topicChannel.Topic, message.ReceivedTopic, "ReceivedTopic not as expected")
```

Comment:
this could be simplified by just range over the channel `topicChannel` so we don't need the `expectedMessages` and `counter`
```go
for message := range topicChannel {
  assert.Equal(t, ....)
}
```

Reply:
Naw, need to range over the expectedMessages to know when to stop, otherwise it would never stop waiting on the channel